### PR TITLE
Custom metrics and some refactory

### PR DESCRIPTION
**Pull request by VulgarnyKarlson**
The problem: pm2/io not provides flexible metrics , u can just use only Name for metrics
in big projects you have much more metrics, this will help

Now u can use custom metrics ( Counters available now only)

```
const { incrementMetric } = require('pm2-metrics');
incrementMetric("metricName", {
  firstKey: "value1",
  secondKey: "value2",
})
incrementMetric("metricName", {
  firstKey: "value2",
  secondKey: "value2",
})
```

in output u will have smth like this:
```
metricName{firstKey="value1",secondKey: "value2"} 1
metricName{firstKey="value2",secondKey: "value2"} 1
```

### CODE REVIEW

**Comment by saikatharryc** [comment]
@VulgarnyKarlson is the final response format is different than previous?

**Comment by VulgarnyKarlson** [comment]
@saikatharryc the final response format is same as Prometheus , the problems was : pm2/io metrics just not giving a chance make metrics with custom labelnames

**Comment by VulgarnyKarlson** [comment]
In next  update i will include another type of metrics and methods , this is just an example (today)

**Comment by VulgarnyKarlson** [comment]
Update:
```
const metrics = require('pm2-metrics')();
metrics({ 
    type: "Counter", 
    method:"inc", 
    createParams: { 
      name: "serverResponses", 
      help: "This is help text of responses", 
      labelNames: ["code", "serverName"] 
    }
  }, [{ code : "400", serverName:"server1.example.com" }, 15]
);
// In output u will have 
/*
  # HELP serverResponses This is help text of responses
  # TYPE serverResponses counter
  serverResponses{code="400",serverName="server1.example.com"} 15
 */
```
For more help u can look on [prom-client](https://github.com/siimon/prom-client)

**Comment by saikatharryc** [comment]
@VulgarnyKarlson  seems like there is lots of changes although i feel if we can accommodate this changes, it will help lots of people.
I was trying to run using `pm2` with your pr, it seems there is no `exporter.js` file inside the `src` folder, can you have a look into this may be?

**Comment by VulgarnyKarlson** [comment]
If u wanna start locally - use ecosystem.config.js

File: ecosystem.config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
   apps : [{
     name: 'PM2 Exporter',
-    script: 'exporter.js',
+    script: 'src/exporter.js',
```
Comment (saikatharryc) [inline ecosystem.config.js]: There is no file named `exporter.js` inside `src` folder.

**Comment by VulgarnyKarlson** [comment]
Oh , my fault , forget to change on index.js :) that’s main importer now

**Comment by saikatharryc** [comment]
On `pm2 start ecosystem.config.js ` 
it gives something like this:
```

**Comment by VulgarnyKarlson** [comment]
You must have env : "pmx_module" to start this , fixed in ecosystem.config.js 
```
if (process.env.pmx_module) {
  const { exporter } = require('./http-server');
  exporter();
} else {
  module.exports = require('./custom-metrics');
}```

**Comment by VulgarnyKarlson** [comment]
up

**Pull request closed without merging**